odhcp6c: Request S46/AFTR DHCPv6 options when packages are installed

detobate opened the pull request:
openwrt/odhcp6c#89 changes the default behaviour of odhcp6c to no longer request S46 options.

This complementary PR updates the map and ds-lite packages to create new logical interfaces upon package install, that in turn instructs odhcp6c to request the required options.

Signed-off-by: Richard Patterson <richard@helix.net.nz>

r-pats committed:
odhcp6c: Request S46 OROs when a MAP-T/E logical interface exists

This is required as the behaviour of odhcp6c is changing to not request these OROs by default.

This commit should be merged along with commits to the map and ds-lite packages, that add configuration to requests the OROs when the packages are installed.

Signed-off-by: Richard Patterson <richard@helix.net.nz>

r-pats committed:
map: Configure a logical MAP-T/E WAN interface when package is installed

This is required as the behaviour of odhcp6c is changing to not request these OROs by default.

Signed-off-by: Richard Patterson <richard@helix.net.nz>

r-pats committed:
ds-lite: Configure a logical DS-Lite WAN interface when package is installed

This is required as the behaviour of odhcp6c is changing to not request the AFTR ORO by default.

Signed-off-by: Richard Patterson <richard@helix.net.nz>

jow- commented:
The odhcp6c handler should automatically request those DHCPv6 options when map and ds-liste packages are installed, e.g. by checking for the existence of `/lib/netifd/proto/map.sh` and `/lib/netifd/proto/dslite.sh` respectively.

Additionally new options could be introduced to forcibly suppress requesting those options for those rare cases where it might be desired.

This way things continue to work as expected while avoiding MAP or DS-Lite requests when the corresponding packages are not installed.

Automatically altering the network configuration upon package install is not robust enough, you cannot assume a certain configuration layout or the existence of a `wan6` interface. Also the network config might not even be in a state where you could change it (e.g. has intentionally not-yet-applied changes that would cut off access when the config is reloaded or it is currently containing syntax errors because someone has it open in an editor and did save an intermediate incomplete state etc.)

detobate commented:
Thanks for the review and feedback @jow-, I was also worried about the presumption of a wan6 interface.
I was however guided to use uci-defaults by @dedeckeh 

The only thing I'd say about using configuration rather than inference, is that we can further improve support (later) by allowing them to be individually toggle-able, and introducing preference too.
e.g. "try MAP-T first, then MAP-T, and fall back to DS-Lite".  

Although I suppose there's nothing stopping us from doing either of those things on top of the default inference that enables them.

Happy to update the PR to infer though, as you suggest.  Thoughts @dedeckeh ?

r-pats committed:
odhcp6c: Request S46 OROs when the map or ds-lite packages are installed
This is required as the behaviour of odhcp6c is changing to not request these OROs by default.

Signed-off-by: Richard Patterson <richard@helix.net.nz>

r-pats committed:
odhcp6c: Request S46 OROs when the map or ds-lite packages are installed
This is required as the behaviour of odhcp6c is changing to not request these OROs by default.

Signed-off-by: Richard Patterson <richard@helix.net.nz>

detobate commented:
OK, I've gone the route of using ubus to check the handlers.  
Initial testing in my lab environment seems to behave as expected

Headcrabed commented:
https://github.com/openwrt/odhcp6c/pull/89 Is now merged, could anyone have a look at this PR?

hauke commented:
You have to update odhcp6c in main branch too. Currently OpenWrt main branch is still on a version from 2024-09-25.

Headcrabed commented:
@hauke I created a seperate bump PR for odhcpv6 here: https://github.com/openwrt/openwrt/pull/17974

Headcrabed commented:
@detobate @r-pats ping

r-pats commented:
Looks like you have odhcp6c under control with the other PR? 
The Test Formalities logs have expired, so I'll have to re-run and try to fix what pops up again.

r-pats commented:
Can someone trigger a re-run of the Test Formalities, or should I just do a bump commit?

Headcrabed commented:
@hauke 

> Can someone trigger a re-run of the Test Formalities, or should I just do a bump commit?

Can you do this? @hauke

Headcrabed commented:
@hauke ping

Headcrabed commented:
> Can someone trigger a re-run of the Test Formalities, or should I just do a bump commit?

@r-pats Seems bump commit would automatically trigger it.

Headcrabed commented:
@r-pats Ping

Headcrabed commented:
@r-pats ping again

Headcrabed commented:
Seems this MR currently blocks bumping odhcpv6 version...

r-pats committed:
odhcp6c: Request S46 OROs when the map or ds-lite packages are installed
This is required as the behaviour of odhcp6c has changing to not request these OROs by default.

Signed-off-by: Richard Patterson <richard@helix.net.nz>

r-pats commented:
> @hauke "Test formalities" test case somehow disappeared, what happened?

I think it needs approval before Test Formalities is run again?

r-pats commented:
If it's easier, I can close it and open a new PR?

Headcrabed commented:
> If it's easier, I can close it and open a new PR?

You can try to open one, since you can always just close that XD

Headcrabed commented:
@r-pats Besides if you want to open a new MR, please link to my odhcpv6 version bump MR.

Headcrabed commented:
@hauke ping for trigger test cases

r-pats commented:
> @r-pats Besides if you want to open a new MR, please link to my odhcpv6 version bump MR.

Just as a comment in the new MR?

r-pats commented:
OK replaced with https://github.com/openwrt/openwrt/pull/18351
Let's see how I did.
